ui/src/app/base/validation.ts

```diff
@@ -1,3 +1,10 @@
+/**
+ * Validate domain name e.g. "www.example.com"
+ * XXX This should be updated as it currently allows e.g. "www.example.com."
+ * https://github.com/canonical-web-and-design/maas-ui/issues/2755
+ */
+export const DOMAIN_NAME_REGEX = /^([a-z\d]|[a-z\d][a-z\d-.]*[a-z\d])*$/i;
```

Comment:
@ClementChaumel I've just moved that domain regex here, which houses a few other regexes